models/core/btc_bridge.sql

```diff
@@ -0,0 +1,43 @@
+{{ 
+    config(
+        materialized='incremental', 
+        unique_key= 'log_id',
+        incremental_strategy = 'delete+insert',
+        tags=['core'],
+        cluster_by = ['block_timestamp']
+    ) 
+}}
+-- Collateral is required to launch a vault
+with vaults as (
+
+  select 
+    distinct event_inputs:_to::string as vault_address
+  from {{ ref('logs') }}
+  where 
+    evm_contract_address = '0xdc54046c0451f9269fee1840aec808d36015697d' -- HMY_ONE_BTC_CONTRACT
+    and event_name = 'IssueTokens'
+
+)
+
+select 
+  log_id,
+  block_id,
+  tx_hash,
+  block_timestamp,
+  contract_address,
+  from_address,
+  to_address,
+  case 
+    when from_address = '0x0000000000000000000000000000000000000000' then 'issue'
+    when to_address = '0x0000000000000000000000000000000000000000' then 'redeem'
+  end as tx_type,
+  v.vault_address is not null as is_vault,
+  raw_amount
+from {{ ref('transfers') }} as t 
```

Comment:
wrap this in a "final" CTE. extract transfer to its own CTE with the incremental filter.